### PR TITLE
[NCL-7586]: Replace maintenance-mode-related endpoints

### PR DESCRIPTION
**Pull request by akridl**
### Checklist:

* [x] Have you added unit tests for your change?

### CODE REVIEW

**Comment by akridl** [comment]
retest this please

File: common/src/main/java/org/jboss/pnc/common/json/JsonUtils.java [hunk 1]
```diff
@@ -32,4 +32,17 @@ public static String toJson(Object objectToBeMapped) throws JsonProcessingExcept
         return new ObjectMapper().writer().withDefaultPrettyPrinter().writeValueAsString(objectToBeMapped);
     }
 
+    /**
+     * @param key The name of the JSON field.
+     * @param value The value of the JSON field.
+     * @return If the value of the JSON field is null, the method returns: "<key>: null".
+     *         </p>
+     *         Otherwise it returns: "<key>": "<value>".
```
Comment (janinko) [inline common/src/main/java/org/jboss/pnc/common/json/JsonUtils.java:40]: inconsistent. one is wholy in quotes (that are not in the string), the other are not.

File: dto/src/main/java/org/jboss/pnc/dto/notification/GenericSettingNotification.java [hunk 2]
```diff
@@ -18,28 +18,22 @@
...
 public class GenericSettingNotification extends Notification {
 
-    public static final String ANNOUNCEMENT_BANNER_CHANGED = "NEW_ANNOUNCEMENT";
-    public static final String MAINTENANCE_STATUS_CHANGED = "MAINTENANCE_STATUS_CHANGED";
+    public static final String PNC_STATUS_CHANGED = "PNC_STATUS_CHANGED";
 
-    public static GenericSettingNotification newAnnoucement(String message) {
-        return new GenericSettingNotification(ANNOUNCEMENT_BANNER_CHANGED, "{\"banner\": \"" + message + "\"}");
+    public static GenericSettingNotification pncStatusChanged(PncStatus pncStatus) {
+        return new GenericSettingNotification(pncStatus.asDto());
```
Comment (janinko) [inline dto/src/main/java/org/jboss/pnc/dto/notification/GenericSettingNotification.java:30]:
```suggestion
        return new GenericSettingNotification(JsonUtils.toJson(pncStatus));
```

**Review by janinko — CHANGES_REQUESTED** [review]

File: dto/src/main/java/org/jboss/pnc/dto/notification/GenericSettingNotification.java (hunk 2 above)
Comment (akridl) [reply dto/src/main/java/org/jboss/pnc/dto/notification/GenericSettingNotification.java:30]: I was wondering whether this is not what I want, and if you suggest it, turns out that's exactlyy what we want to use! :heart_eyes:

File: common/src/main/java/org/jboss/pnc/common/json/JsonUtils.java (hunk 1 above)
Comment (akridl) [reply common/src/main/java/org/jboss/pnc/common/json/JsonUtils.java:40]: :eagle:

File: rest/src/main/java/org/jboss/pnc/rest/endpoints/internal/PncStatusEndpointImpl.java [hunk 3]
```diff
@@ -42,11 +42,15 @@ public void setPncStatus(PncStatus pncStatus) {
             throw new BadRequestException("Banner cannot be blank.");
         }
 
-        if (pncStatus.getBanner() != null) {
+        if (pncStatus.getBanner() == null) {
+            genericSettingProvider.clearAnnouncementBanner();
+        } else {
             genericSettingProvider.setAnnouncementBanner(pncStatus.getBanner());
         }
```
[the remark this answers is not in the document]
Comment (akridl) [reply rest/src/main/java/org/jboss/pnc/rest/endpoints/internal/PncStatusEndpointImpl.java:49]: Not relevant anymore, resolving this thread

File: common/src/main/java/org/jboss/pnc/common/json/JsonUtils.java [hunk 4]
```diff
@@ -31,5 +31,4 @@ public static <T> T fromJson(String json, Class<T> aClass) throws IOException {
     public static String toJson(Object objectToBeMapped) throws JsonProcessingException {
         return new ObjectMapper().writer().withDefaultPrettyPrinter().writeValueAsString(objectToBeMapped);
     }
-
```
Comment (janinko) [inline common/src/main/java/org/jboss/pnc/common/json/JsonUtils.java:34]: :raised_eyebrow:

**Review by janinko — APPROVED** [review]

File: common/src/main/java/org/jboss/pnc/common/json/JsonUtils.java (hunk 4 above)
Comment (akridl) [reply common/src/main/java/org/jboss/pnc/common/json/JsonUtils.java:34]: CTRL + ALT + L :smile:

**Pull request merged**